pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -91,6 +96,29 @@ func (l *projectLifecycle) sync(key string, orig *v3.Project) error {
 		}
 	}
 
+	// create cluster role for this cluster's templates and templateversions. cluster role here too since we want to grant access to templates resource
+	// which is not namespaced
+	clusterName := orig.Namespace
+	gotCR, err := l.mgr.clusterRoleClient.Get("project-"+clusterName+"-"+orig.Name+"-use-templates-templateversions", metav1.GetOptions{})
+
```

Comment:
we should check if gotCR is not nil + there is no error check being done. As a part of error check see if the error is already exists error, and do create if it is.